Source/ThirdParty/skia/CMakeLists.txt

```diff
@@ -0,0 +1,1019 @@
+set_property(DIRECTORY . PROPERTY FOLDER "skia")
+
+if (SKIA_ROOT_DIR OR NOT "$ENV{SKIA_ROOT_DIR}" STREQUAL "")
+    if (NOT SKIA_ROOT_DIR)
+        set(SKIA_ROOT_DIR "$ENV{SKIA_ROOT_DIR")
+    endif ()
+
+    find_path(SKIA_INCLUDE_DIR
+        NAMES include/private/base/SkFeatures.h
+        HINTS ${SKIA_INCLUDE_DIR} ${SKIA_ROOT_DIR}
+        NO_DEFAULT_PATH
+        REQUIRED
+    )
+    find_library(SKIA_LIBRARY
+        NAMES skia libskia
+        HINTS ${SKIA_LIBRARY_PATH} ${SKIA_ROOT_DIR}/out ${SKIA_ROOT_DIR}/out/Static
+        NO_DEFAULT_PATH
+        REQUIRED
+    )
+    find_path(SKIA_NINJA_RULES
+        NAMES skia.ninja
+        HINTS ${SKIA_LIBRARY_PATH} ${SKIA_ROOT_DIR}
+        PATH_SUFFIXES out/obj out/Static/obj
+        NO_DEFAULT_PATH
+        REQUIRED
+    )
+
+    # Parse the top-level Skia build rules, to get the macros used for
+    # building it because they need to be passed when building WebKit
+    # sources. This is needed because *public* Skia headers use these
+    # macros but the build system does *not* generate a configuration
+    # file.
+    file(STRINGS "${SKIA_NINJA_RULES}/skia.ninja"
+        SKIA_NINJA_RULES_LINES
+        REGEX "^defines[ \t]="
+        NO_HEX_CONVERSION
+    )
+    foreach (line ${SKIA_NINJA_RULES_LINES})
+        string(REGEX MATCH "^defines[ \t]+=[ \t]+(.*)$" _dummy "${line}")
+        separate_arguments(args NATIVE_COMMAND "${CMAKE_MATCH_1}")
+        foreach (arg IN LISTS args)
+            if (arg MATCHES "^-D_?N?DEBUG(=.*)?$")
+                continue ()
+            endif ()
+            if (arg MATCHES "^-DSKIA_IMPLEMENTATION(=.*)?$")
+                continue ()
+            endif ()
+            if (arg MATCHES "^-D(.*)$")
+                list(APPEND SKIA_COMPILE_DEFINITIONS "${CMAKE_MATCH_1}")
+            endif ()
+        endforeach ()
+    endforeach ()
+
+    # Remove the SKIA_IMPLEMENTATION macro from the list.
+    foreach (flag "-DSKIA_IMPLEMENTATION" "-DSKIA_IMPLEMENTATION=1")
+        list(REMOVE_ITEM SKIA_COMPILE_DEFINITIONS "${flag}")
+    endforeach ()
+
+    add_library(Skia UNKNOWN IMPORTED GLOBAL)
+    set_target_properties(Skia PROPERTIES
+        IMPORTED_LOCATION "${SKIA_LIBRARY}"
+        INTERFACE_INCLUDE_DIRECTORIES "${FORWARDING_HEADERS_DIR};${SKIA_INCLUDE_DIR}"
+        INTERFACE_COMPILE_DEFINITIONS "${SKIA_COMPILE_DEFINITIONS}"
+    )
+
+    find_package(EGL REQUIRED)
+    find_package(OpenGLES2 REQUIRED)
+    find_package(Freetype 2.9.0 REQUIRED)
+    find_package(Fontconfig 2.13.0 REQUIRED)
+    find_package(PNG REQUIRED)
```

Comment:
These is already present in `Source/cmake/Options${PORT_NAME}.cmake` so they aren't needed. If a specific library version is needed it should go in there not here.

Reply:
I agree we don't want to find packages twice, because of our policy dependency we can't make skia depend on a different version, so it's ok to have a single find package in the main cmake file for shared dependencies. But I also agree with having find_package where they are actually needed, so what I would do here is:

```
find_package(EGL REQUIRED)
find_package(OpenGLES2 REQUIRED)
```

I don't think we need these for WPE and GTK since epoxy is our dependency, so I would just remove those for now.

```
find_package(Freetype 2.9.0 REQUIRED)
find_package(Fontconfig 2.13.0 REQUIRED)
```

Those are in the main file because WebCore depends on those, but that's not the case when building with skia, so I would move those inside a if USE_CAIRO in the main file and keep them here too.

```
find_package(PNG REQUIRED)
```

This is required by the image decoders that are used when building with skia, so I would remove this from here and just use the one in the main file.

Reply:
Oh, I realized now that this is for the external skia.

Reply:
I'm going to simplify both a bit with the suggestions, thanks to both!

Reply:
Done!